bot/core/paginator.py

```diff
@@ -0,0 +1,421 @@
+                        # If element is an empty dictionary
+                        if element == {}:
+                            raise PaginatorError(
+                                "FieldPaginator argument is a nested dictionary but does not have the mandatory "
+                                "key-value pair".strip()
+                            )
+                        elif len(element.items()) > 2:
+                            raise PaginatorError(
+                                "FieldPaginator argument is supposed to be a dictionary of "
+                                "2 key-value pair (key: value, inline: bool), but found " + len(element.items())
+                            )
+
+                        _embed.add_field(
+                            name=list(element.keys())[0],
+                            value=list(element.values())[0],
+                            inline=True if 'inline' not in element.keys() else bool(element['inline'])
+                        )
+                    else:
+                        raise PaginatorError(
+                            "\n"
+                            "Field paginator nested embed is not accepted\n"
+                            "Eg: FieldPaginator([{'Name': 'John', 'inline': False}, ->Embed().set_footer('Name', 'Sarah')<-])\n"
+                            "Accepted type to create an embed for FieldPaginator (note that this will create a new page):\n"
+                            "FieldPaginator([{'Name': 'John', 'inline': False}], Embed().set_footer('Name', 'Sarah'))"
+                        )
+                self.embeds.append(_embed)
+            else:
+                raise PaginatorError(
+                    "FieldPaginator argument given "
+                    f"at index {index} is invalid.\n"
+                    "The supported types are:\n"
+                    "Embed and list"
+                )
+
+
+class ListPaginator(_BasePaginator):
+    """
+    A class which creates a list of embeds and configures
+    their descriptions according to the parameters.
+
+    Parameters
+    ----------
+        entries : List[str]
+            List of entries to add to the paginator
+            Length of every entry must be lower than
+            the max_size argument.
+        individual_paging : bool
+            Whether to make a new paginator page for every
+            entry.
+        prefix : str
+            A string that is added to the beginning to every
+            paginator page.
+        suffix : str
+            A string that is added to the end of every paginator
+            page.
+        max_size : int
+            Maximum amount of characters allowed per entry.
+    """
+    def __init__(
+        self,
+        entries: t.Optional[t.List[str]] = None,
+        individual_paging=True,
+        prefix='',
+        suffix='',
+        max_size=2000
+    ) -> None:
+
+        entries = entries if entries is not None else []
```

Comment:
While there is technically nothing wrong with your syntax, there is no need for ternary conditional here since it only resets the variable to itself or to the preferred value, using regular if statement would be preferred as it is more commonly understood hence making it easier to read and doesn't waste time on re-setting the variable to itself.
```suggestion
        if entries is None:
            entries = []
```